### ISSUE TITLE
Using synchronize_structure fails when replicating from 9.4 to 9.5

### ISSUE THREAD

**Issue by simaofreitas** [2ndQuadrant/pglogical#35]
Hi

I'm trying to replicate from a 9.4 server to a 9.5 server.
I created the nodes on both provider & subscriber and also added all tables from the public schema to the default replication set.

When creating the subscription with:
```
LECT pglogical.create_subscription(
    subscription_name := 'subscriptionname',
    provider_dsn := 'host=providerhost port=5432 dbname=dbname user=replicator password=teste',
    synchronize_structure := true
);
```

The command fails because it tries to create the `pglogical_origin` extension in the 9.5 server (that is included in the generated dump).

This extension is not available in 9.5 servers, making the `synchronize_structure` non usable when creating the subscription.

Any clues or workarounds?

Thanks

**Comment by PJMODOS** [comment]
Yes, I know about this issue, the only workaround I know so far is to use schema only pg_dump manually and skip the synchronize_structure.

I think we should revert the change that requires pglogical_origin to be always installed since it's really only needed on 9.4 subscriber and it breaks the online upgrade scenario for 9.4 to 9.5.

**Comment by mafonso** [comment]
I just bumped into this very same issue. 

```ERROR,58P01,"could not open extension control file ""/usr/pgsql-9.5/share/extension/pglogical_origin.control"": No such file or directory",,,,,,"CREATE EXTENSION IF NOT EXISTS pglogical_origin WITH SCHEMA pglogical_origin;`

**Comment by ringerc** [comment]
Or just provide an empty dummy pglogical_origin that we install on 9.5​

**Comment by mafonso** [comment]
@ringerc you mean something like adding a .control file with empty module_pathname?
Something like this?  Or is there a actual dummy implementation of such module?  I could not find one...

```
comment = 'PostgreSQL Logical Replication Origin Tracking Dummy Emulation for 9.5'
default_version = '1.0.0'
module_pathname = ''
relocatable = false
schema = pglogical_origin
```

**Comment by ringerc** [comment]
There isn't a dummy implementation now, but yes, that's what I'm thinking.
Install a dummy module if we're on 9.5 to make pg_restore happy.

-- 
 Craig Ringer                   http://www.2ndQuadrant.com/
 PostgreSQL Development, 24x7 Support, Training & Services

**Comment by ringerc** [comment]
Empty module_pathname wouldn't do it. We'd have a different control file
we'd install for 9.5 servers that clearly said it was a dummy, and an empty
sql file to run.

**Comment by mafonso** [comment]
I kinda made it work with the empty `module_pathname` and then an sql file with just select 1;
I allow me to create the extension manually so that pgrestore skips over it.

````"CREATE EXTENSION",2016-08-23 04:23:43 CDT,6/1425,2538,NOTICE,42710,"extension ""pglogical_origin"" already exists, skipping",,,,,,,,,"pg_restore"`

**Comment by ringerc** [comment]
I think this is what @PJMODOS  meant to fix with https://github.com/2ndQuadrant/pglogical/commit/5edcc89d5c8e115c40b5e7141b682d531e4a6fea, which in turn resulted in #45 . He's doing a different fix.

**Comment by PJMODOS** [comment]
should be fixed in 1.2.1
